var/spack/repos/builtin/packages/mesa/package.py

```diff
@@ -114,7 +114,6 @@ class Mesa(MesonPackage):
         depends_on("libllvm@6:")
         depends_on("libllvm@:11", when="@:20")
         depends_on("libllvm@:12", when="@:21")
-        depends_on("libllvm@:15", when="@:22.1")
```

Comment:
Isn't this still true? Why did you remove it?

Reply:
When we set depends_on("libllvm@:14", when="@:22.1") it is concretizing for hip@5.3.0 with mesa@main instead of mesa@22.1.2 as given below.
[hip_spec_with_mesa_depends_on_libllvm14.txt](https://github.com/spack/spack/files/9992889/hip_spec_with_mesa_depends_on_libllvm14.txt)

I verified the above change (without adding any additional dependency with libllvm). 
hip@5.3.0 is getting concretized with mesa version to 22.1.2. 
But we need the patch to work with the llvm-amdgpu@5.3.0

Reply:
ah, you could just remove the `when="@:22.1"` and that will say all versions of mesa currently require `llvm@:14`, which is true. Then it won't allow building mesa with `hip@5.3.0` and require some other GL provider.

Again, I would really prefer to not patch this unless we absolutely need to have a spack built ROCm 5.3.0 stack from spack, just having the updates for configuring the bulid environments for the dependent packages using externals is enough for most users who are building on systems with the ROCm stack already installed from binaries.

Reply:
ROCm 5.3.0 provides llvm15. Gallivm coroutine is disabled in mesa upstream version for llvm-15. Until mesa release is available with the changes for Gallivm coroutine patch is required with ROCm 5.3.0. This patch is restricted to libllvm-15 (ROCm 5.3.0) [fc4b159](https://github.com/spack/spack/pull/33320/commits/fc4b159cd4c1e597004d996c974fb2c8934bd7bb)